src/ucp/core/ucp_ep.c

```diff
@@ -3473,7 +3474,6 @@ void ucp_ep_req_purge(ucp_ep_h ucp_ep, ucp_request_t *req,
         ucp_request_complete_tag_recv(req, status);
     } else if (req->flags & UCP_REQUEST_FLAG_RNDV_FRAG) {
         ucs_assert(req->flags & UCP_REQUEST_FLAG_SUPER_VALID);
-        ucs_assert(req->send.ep == ucp_ep);
```

Comment:
why?

Reply:
see PR description, this is `recv` request, we can not access its `send` part

Reply:
how can it be recv request if we removed it from the pending queue?

Reply:
this is a super request of RTR, hierarchy of completions is 
```
rtr_req -> frag_req -> recv_req
(send)     (recv)      (recv)      
```

Reply:
i see
https://github.com/openucx/ucx/blob/dfdd4f4dd733d77c38a70f8de4e3366f8d4c060d/src/ucp/rndv/rndv.c#L1378